src/SyntheticDataWorkValidator.sol

```diff
@@ -84,6 +86,37 @@ contract SyntheticDataWorkValidator is IWorkValidation {
         return (info.provider, info.nodeId);
     }
 
+    function softInvalidateWork(uint256 poolId, bytes calldata data) external returns (address, address, uint256) {
+        require(msg.sender == computePool, "Unauthorized");
+        require(data.length >= 64, "Data too short for soft invalidation");
+
+        // Decode both workKey and workUnits for soft invalidation
+        bytes32 workKey;
+        uint256 workUnits;
+        assembly {
+            workKey := calldataload(data.offset)
+            workUnits := calldataload(add(data.offset, 32))
+        }
+
+        require(poolWork[poolId].workKeys.contains(workKey), "Work not found");
+        require(!poolWork[poolId].invalidWorkKeys.contains(workKey), "Work already invalidated");
+        require(
+            block.timestamp - poolWork[poolId].work[workKey].timestamp < workValidityPeriod,
+            "Work invalidation window has lapsed"
+        );
+
+        WorkInfo memory info = poolWork[poolId].work[workKey];
+
+        // Move work from valid to invalid (same as hard invalidation)
+        poolWork[poolId].invalidWorkKeys.add(workKey);
+        poolWork[poolId].workKeys.remove(workKey);
+
+        emit WorkRemoved(poolId, info.provider, info.nodeId, workUnits);
```

Comment:
could we also include the `workKey` here in the event -- it simplifies handling on the platform-side